### PR TITLE
[VisLib] Replace legend color palette with OUI color palette

### PR DESCRIPTION
**Pull request by joshuarrrr**
### Description

Replace hard-coded palette (of 8 colors with 7 variations each) with rotations from euiPaletteColorBlind (of 10 colors with 7 variations each).

Note that this means an increase in the total number of colors available to select from the legend picker (56->70) and a wider color picker component.

We also experimented with using [`colorPalette()`](https://oui.opensearch.org/1.0/#/utilities/color-palettes) with each of the 10 base colors, but it's designed to make quantitative palettes that are fully desaturated at one end, so the lightest values were much too light.

### Issues Resolved

Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4321

## Screenshot

Legend, with default base colors

Screen Shot 2023-06-22 at 2 01 54 PM

Single color variants:
Screen Shot 2023-06-22 at 2 02 24 PM

Darkest shades:
Screen Shot 2023-06-22 at 2 03 26 PM

Lightest shades:

Screen Shot 2023-06-22 at 2 03 59 PM

Heatmap:
Screen Shot 2023-07-18 at 11 52 50 AM

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/vis_type_vislib/public/vislib/components/legend/_legend.scss [hunk 1]
```diff
@@ -84,7 +84,7 @@ $visLegendLineHeight: $euiSize;
 }
 
 .visLegend__valueColorPicker {
-  width: ($euiSizeL * 8); // 8 columns
+  width: ($euiSizeL * 10); // 10 columns
```
Comment (manasvinibs) [inline src/plugins/vis_type_vislib/public/vislib/components/legend/_legend.scss:87]: Is this 10 because that's the default size of oui palette?

**Review by BSFishy — APPROVED** [review]

File: src/plugins/vis_type_vislib/public/vislib/components/legend/legend_item.tsx [hunk 2]
```diff
@@ -42,9 +42,14 @@ import {
...
   EuiButtonGroupOptionProps,
+  euiPaletteColorBlind,
 } from '@elastic/eui';
 
-import { legendColors, LegendItem } from './models';
+import { LegendItem } from './models';
+
+// starting from the default categorical colors, we generate 6 additional variants,
+// 3 lighter and 3 darker
+const legendColors = euiPaletteColorBlind({ rotations: 7, direction: 'both' });
```
Comment (manasvinibs) [inline src/plugins/vis_type_vislib/public/vislib/components/legend/legend_item.tsx:52]: Why do we need to set `direction` here?

File: src/plugins/vis_type_vislib/public/vislib/components/legend/models.ts [hunk 3]
```diff
@@ -34,62 +34,3 @@ export interface LegendItem {
 }
 
 export const CUSTOM_LEGEND_VIS_TYPES = ['heatmap', 'gauge'];
```
Comment (manasvinibs) [inline src/plugins/vis_type_vislib/public/vislib/components/legend/models.ts:36]: Do we also want to add `heatmap` screenshot with updated legend palette?

**Review by manasvinibs — APPROVED** [review]

File: src/plugins/vis_type_vislib/public/vislib/components/legend/legend_item.tsx (hunk 2 above)
Comment (BSFishy) [reply src/plugins/vis_type_vislib/public/vislib/components/legend/legend_item.tsx:52]: By default, `direction` is `'lighter'`, which means that we wouldn't get the darker colors.

Comment (manasvinibs) [reply src/plugins/vis_type_vislib/public/vislib/components/legend/legend_item.tsx:52]: Ah I see! So how does it gets divided to 3 lighter and 3 darker?

Comment (BSFishy) [reply src/plugins/vis_type_vislib/public/vislib/components/legend/legend_item.tsx:52]: `direction: 'both'` basically generates a palette for each of the 10 base colors that goes from white to the color to black and chops off white and black. Setting `direction` to either `'lighter'` or `'darker'` basically chops off the lighter or darker end of the palette, so setting it to `'both'` just keeps the whole palette.

Comment (manasvinibs) [reply src/plugins/vis_type_vislib/public/vislib/components/legend/legend_item.tsx:52]: Makes sense!

File: src/plugins/vis_type_vislib/public/vislib/components/legend/_legend.scss (hunk 1 above)
Comment (joshuarrrr) [reply src/plugins/vis_type_vislib/public/vislib/components/legend/_legend.scss:87]: yep - see https://oui.opensearch.org/1.0/#/utilities/color-palettes

File: src/plugins/vis_type_vislib/public/vislib/components/legend/models.ts (hunk 3 above)
Comment (joshuarrrr) [reply src/plugins/vis_type_vislib/public/vislib/components/legend/models.ts:36]: Sure, I'll add that - I think it renders the same.

**Comment by joshuarrrr** [comment]
I still need to update some unit and functional tests that tested the old hard-coded legend color values.

**Comment by manasvinibs** [comment]
> I still need to update some unit and functional tests that tested the old hard-coded legend color values.

Ya I see CIgroup 4 tests are failing for some hard coded colors. I have fixed this as part of my PR https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4348 But you can go ahead and fix it here, whichever merges first we can rebase.

**Review by BSFishy — APPROVED** [review]

**Review by manasvinibs — APPROVED** [review]

**Pull request merged**